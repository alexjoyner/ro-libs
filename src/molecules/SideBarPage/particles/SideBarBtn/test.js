import React from 'react';
import { mount } from 'enzyme';
import { SideBarBtn } from './';

describe('SideBarBtn component', () => {
  let component;
  beforeEach(() => {
    component = mount(<SideBarBtn />);
  });
  it('Should render without exploding', () => {
    expect(component).toBeDefined();
  });
  it('Should match snapshot', () => {
    expect(component).toMatchSnapshot();
  });
  it('should default to correct position', () => {
    expect(component).toHaveStyleRule('left', '7px');
  });
  it('should accept shown prop and change style', () => {
    const shownTest = mount(<SideBarBtn shown />);
    expect(shownTest).toHaveStyleRule('left', '260px');
  });
});
