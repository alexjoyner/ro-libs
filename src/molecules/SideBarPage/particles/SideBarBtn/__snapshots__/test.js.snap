// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBarBtn component Should match snapshot 1`] = `
.c1 {
  -moz-appearance: none;
  -webkit-appearance: none;
  background: none;
  border: none;
  outline: none;
  font-size: inherit;
}

.c2 {
  background-color: #b2bec3;
  color: black;
  border: 1px solid #b2bec3;
  display: inline-block;
  outline: none;
  cursor: pointer;
  font-size: 1rem;
  padding: .375rem .75rem;
  text-align: center;
  line-height: 1.5;
  -webkit-transition: 0.1s linear;
  transition: 0.1s linear;
  box-shadow: 0 1px 2px 0 rgba(0,0,0,0.2);
}

.c2 svg {
  vertical-align: middle;
}

.c2:hover {
  box-shadow: 0 2px 4px 0 rgba(0,0,0,0.2),0 3px 10px 0 rgba(0,0,0,0.19);
}

.c2:active {
  box-shadow: none;
}

.c0 {
  margin-left: 0;
  -webkit-transition: 100ms linear;
  transition: 100ms linear;
  position: fixed;
  margin-top: 2px;
  left: 7px;
}

<Styled(Button)>
  <StyledComponent
    forwardedClass={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": undefined,
        "componentStyle": ComponentStyle {
          "componentId": "sc-htpNat",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "
  margin-left: 0;
  transition: 100ms linear;
  position: fixed;
  margin-top: 2px;
  ",
            [Function],
            "; 
",
          ],
        },
        "displayName": "Styled(Button)",
        "render": [Function],
        "styledComponentId": "sc-htpNat",
        "target": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
        Symbol(Symbol.hasInstance): [Function],
      }
    }
    forwardedRef={null}
  >
    <Button
      className="c0"
      color="default"
      ghost={false}
      size="default"
    >
      <styled.button
        className="c0"
        type="button"
      >
        <StyledComponent
          className="c0"
          forwardedClass={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": undefined,
              "componentStyle": ComponentStyle {
                "componentId": "sc-bdVaJa",
                "isStatic": true,
                "lastClassName": "c1",
                "rules": Array [
                  "
  -moz-appearance: none;
  -webkit-appearance: none;
  background: none;
  border: none;
  outline: none;
  font-size: inherit;
",
                ],
              },
              "displayName": "styled.button",
              "render": [Function],
              "styledComponentId": "sc-bdVaJa",
              "target": "button",
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
          type="button"
        >
          <button
            className="c0 c1"
            type="button"
          >
            <styled.span
              color="default"
              ghost={false}
              size="default"
              tabIndex={-1}
            >
              <StyledComponent
                color="default"
                forwardedClass={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": undefined,
                    "componentStyle": ComponentStyle {
                      "componentId": "sc-bwzfXH",
                      "isStatic": false,
                      "lastClassName": "c2",
                      "rules": Array [
                        "
    background-color: ",
                        [Function],
                        ";
    color: ",
                        [Function],
                        ";
    border: 1px solid ",
                        [Function],
                        ";
    display: inline-block;
    outline: none;
    cursor: pointer;
    ",
                        [Function],
                        "; 
    text-align: center;
    line-height: 1.5;
    transition: 0.1s linear;
    box-shadow: 0 1px 2px 0 rgba(0, 0, 0, 0.2);
    svg {
      vertical-align: middle;
    }
    &:hover {
      box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.2), 0 3px 10px 0 rgba(0, 0, 0, 0.19);
    } 
    &:active {
      box-shadow: none;
    }
    ",
                        [Function],
                        ";
  ",
                      ],
                    },
                    "displayName": "styled.span",
                    "render": [Function],
                    "styledComponentId": "sc-bwzfXH",
                    "target": "span",
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
                ghost={false}
                size="default"
                tabIndex={-1}
              >
                <span
                  className="c2"
                  color="default"
                  size="default"
                  tabIndex={-1}
                />
              </StyledComponent>
            </styled.span>
          </button>
        </StyledComponent>
      </styled.button>
    </Button>
  </StyledComponent>
</Styled(Button)>
`;
